pallets/rmrk-core/src/functions.rs

```diff
@@ -330,19 +337,20 @@ where
 	fn nft_mint(
 		sender: T::AccountId,
 		owner: T::AccountId,
+		nft_id: NftId,
 		collection_id: CollectionId,
 		royalty_recipient: Option<T::AccountId>,
 		royalty_amount: Option<Permill>,
 		metadata: StringLimitOf<T>,
 		transferable: bool,
-		resources: Option<BoundedResourceTypeOf<T>>,
+		resources: Option<BoundedResourceInfoTypeOf<T>>,
 	) -> sp_std::result::Result<(CollectionId, NftId), DispatchError> {
-		let nft_id = Self::get_next_nft_id(collection_id)?;
+		ensure!(Nfts::<T>::get(collection_id, nft_id).is_none(), Error::<T>::NftAlreadyExists);
```

Comment:
Use `nft_exists` instead to check. We can likely refactor the function to return an Error.

Reply:
Thanks! Good point

Reply:
Updated, won't change it for now to return an error as it's used for other purposes for equippables